### ISSUE TITLE
builtins.match: \s and \S do not work in 1.12

### ISSUE THREAD

**Issue by volth** [NixOS/nix#1331]
```builtins.match ''\s+(\S+)\s+'' " foo "``` works in ```nix-1.11.8``` but fails with ```error: regex_error``` in ```nix-1.12pre5122_c60715e```

**Comment by edolstra** [comment]
This is probably a consequence of the switch to `std::regex` in b05b98df7544d02387f583ca5434f33f3e9cb471.

**Comment by edolstra** [comment]
`builtins.match` supports POSIX extended regular expressions, which AFAIK don't support `\s` and `\S`. Apparently the libc RE functions (which we used prior to b05b98d) do support them, but this is undocumented (see https://linux.die.net/man/7/regex) and not portable. For instance, on OS X, your example returns `null`.

I've added a note to the manual documenting that `builtins.match` support POSIX extended REs.
